pycrest/eve.py

```diff
@@ -159,7 +159,7 @@ def __init__(
         if user_agent is None:
             user_agent = "PyCrest/{0} +https://github.com/pycrest/PyCrest"\
                 .format(version)
-        if transport_adapter is not None:
+        if transport_adapter is not None and isinstance(transport_adapter, HTTPAdapter):
```

Comment:
Could this also be `if transport_adapter and isinstance...`  since you would need something to not be None when isinstance checks anyway?